docs/Getting Started/Integration Basics/index.md

```diff
@@ -1236,3 +1236,201 @@ The `SDLSessionBean` should be inside a Java package other than the default pack
+
+#### NodeJS Setup
+This build allows you to create apps that run on your computer. [NodeJS](https://nodejs.org/en/) and npm are the tools you will need to integrate this library into what will be your app.
+
+In a new directory, save your `SDL.min.js` build there, and then create a new file in the same directory named `index.js`. Make the contents of that file the following:
+
+```js
+const SDL = require('./SDL.min.js');
+console.log(SDL);
+```
+
+Finally, run the file: `node index.js`. You should see the SDL library version be printed and the imported SDL object. If so, then you have successfully set up the SDL library!
+
+#### Basic Configuration
+The rest of this guide will cover topics that apply to both library builds. Specific code snippets for each environment will be shown when applicable.
+
+In order to correctly connect to an SDL enabled head unit, developers need to create a configuration object to pass into an instance of the `SdlManager` class and then start it. This configuration object is the `AppConfig`, which requires a `LifecycleConfig` object. `LifecycleConfig` contains the majority of the settings a developer would need to set in order for the app to be operable. At the very least, you will want to use the following methods to get started: `setAppId, setAppName, setLanguageDesired, setAppTypes, setTransportConfig`. These configuration objects support method chaining, which allow you to create the following example configuration object:
+
+```js
+const lifecycleConfig = new SDL.manager.LifecycleConfig()
+    .setAppId('hello-js')
+    .setAppName('Hello JS')
+    .setLanguageDesired(SDL.rpc.enums.Language.EN_US)
+    .setAppTypes([
+        SDL.rpc.enums.AppHMIType.DEFAULT,
+    ]);
+```
+
+#### Transport Configuration
+The transport configuration will depend on the environment you're using. For instance, you can make a TCP connection with NodeJS, but not in the browser. Setting up the transport correctly will be covered in the next section. See below for example transport configurations.
+
+```js
+// Example configurations to be completed by the developer
+
+// NodeJS TCP
+lifecycleConfig.setTransportConfig(new SDL.transport.TcpClientConfig(HOST, PORT));
+
+// NodeJS WebSocket Server
+lifecycleConfig.setTransportConfig(
+    new SDL.transport.WebSocketServerConfig(
+        wsClient, // The developer is responsible for providing incoming client connections
+        CONNECTION_LOST_TIMEOUT // connection timeout in milliseconds
+    )
+);
+
+// Plain JS WebSocket Client
+lifecycleConfig.setTransportConfig(new SDL.transport.WebSocketClientConfig(HOST, PORT));
+```
```

Comment:
```suggestion
##### Node.js TCP Transport
```js
lifecycleConfig.setTransportConfig(new SDL.transport.TcpClientConfig(HOST, PORT));
```

##### Node.js WebSocket Server
```js
lifecycleConfig.setTransportConfig(
    new SDL.transport.WebSocketServerConfig(
        wsClient, // The developer is responsible for providing incoming client connections
        CONNECTION_LOST_TIMEOUT // connection timeout in milliseconds
    )
);
```

##### Vanilla JS WebSocket Client
```js
lifecycleConfig.setTransportConfig(new SDL.transport.WebSocketClientConfig(HOST, PORT));
```
```